### DISCUSSION TITLE
Help with connection refused (os error 111)

### DISCUSSION THREAD

**Discussion by darkufo** [rustdesk/rustdesk#2938] [Q&A] [answered]
Hi,

I've recently setup 2 Fedora 37 computers both on the same network.

When I try to access them via RustDesk locally (using the local IP) I'm getting 

connection refused (os error 111)

I'm not sure where to go to check/fix this.

FYI: Using the same computers (dual boot to Windows 11). RustDesk works fine on both computers. 

So I suspect it's a Fedora issue and maybe something to do with the firewall. 

Any help is much appreciated.

NOTE: If I use Fedora, I can connect to my Windows machine using RustDesk. So I suspect both my Fedora machines are refusing connections coming in. But I've no idea how to allow RustDesk

**Comment by rustdesk** [comment]
https://www.errorsolutions.tech/error/linux-error-111-connection-refused/

**Reply by darkufo** [reply]
Thank for the link.

I'm afraid I'm a bit confused about what to actually do.

It says for the 4th Step 

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST= "input your host name" )(PORT=1521)))127.0.0.1 localhost localhost.localdomain localhost4 localhost4.localdomain4
::1 localhost localhost.localdomain localhost6 localhost6.localdomain6
192.168.1.1 "input your host name" oracle

What is that 192.168.1.1 ip address? Is that the IP address of the machine I am using? or the machine I want to connect to?
Where do I get the hostname from? 
What is the word oracle? Is that something I need to set?

Sorry but this is very confusing for me. When I used RustDesk for windows it just worked beautifully :)

Anyone who can help walk a beginner through this would be much appreciated.

**Reply by serious-angel** [reply]
> Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST= "input your host name" )(PORT=1521)))127.0.0.1 localhost localhost.localdomain localhost4 localhost4.localdomain4 ::1 localhost localhost.localdomain localhost6 localhost6.localdomain6 192.168.1.1 "input your host name" oracle
> 
> Sorry but this is very confusing for me...

It's not confused for you only, @darkufo, since it's an utter nonsense in general likely generated by some awful LLM and got published for unknown reason by someone who has absolutely no idea what they are talking about.

This is not a solution. It's a mess that does not deserve your time, and I have no idea why someone at @rustdesk considered it with enough sense to comment with. Perhaps, they didn't even read it, and that suggests the attitude towards the issues in general, I do believe.

**Comment by thespoilergirl** [comment]
Having a similar issue.

I have a Fedora Workstation with RustDesk installed. 

I can connect to my 4 Windows 11 Workstations from the Fedora Workstation via RustDesk. 
I cannot connect to my other Fedora Workstation via RustDesk. Getting OS ERROR 111

I can ping the Fedora Workstation, and I can SSH into into but RustDesk does not want to connect.

All the devices are on the same local network and I am trying to connect via Local IP addresses.

Fedora 37 is running SELinux if that makes a difference

**Comment by thespoilergirl** [comment, marked as answer]
I managed to figure this out. I had forgotten it was a new install

1) Make sure your Ports are open in the firewall
2) Make sure you Enable Direct Connection Access
3) Make sure you Set a Password. It's random by default

**Comment by darkufo** [comment]
That worked. Thank you

**Comment by demuxer86** [comment]
I am getting this same problem.   I have been running on 1.9 for a long time.  Everything worked fine.    Server is self hosted in a docker compose container with Portainer.   I upgraded that to rustdesk/rustdesk-server-s6.

In both windows and fedora 38 (cinnamon), the rustdesk apps are upgraded to the latest nightly.   I rebooted both machines multiple times, and both machines are fully up to date. 

Fedora 38 is running on Proxmox with firewall disabled and SElinux disabled as well. 

I *can* connect to the windows machine from the fedora.  That works fine.   But when I try to use windows to fedora, it says connection refused.   

Where should I look to find the appropriate logs?  I can't seem to find them. 

also, i can connect to the windows machine from my android phone OK too.  im typing this edit with my phone right now.   So the problem seems clearly with Fedora 38 Cinnamon.

**Comment by serious-angel** [comment]
You might have a case of a proxy within proxy, and therefore, please check the following: https://github.com/shadowsocks/shadowsocks-rust/issues/1104#issuecomment-3223618914
